services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,19 +43,44 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  
+  if (event == NULL) {
+    return ERR_CODE_INVALID_ARG;
+  }
+  xQueueSend(thermalMgrQueueHandle, event, 0);
   return ERR_CODE_SUCCESS;
+  
 }
 
 void osHandlerLM75BD(void) {
+  
   /* Implement this function */
+  
+  float *temp;
+  readTempLM75BD(LM75BD_OBC_I2C_ADDR, temp);
+  if (*temp > 75.0) {
+    overTemperatureDetected();
+  }
+  else {
+    safeOperatingConditions();
+  } 
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  
+  float *temp;
```

Comment:
You're not actually allocating any memory to store the temperature value. You're just allocating memory to store a pointer and passing the pointer into `readLM75BD`. This could result in the program accessing a random part of memory (and potentially causing a segmentation fault) when it dereferences that pointer.